Add database

EvgenyAlexandrovichM opened the pull request:
12 спринт.1 Ревью. Подключение базы данных и реализация работы с ней.
Реализованы паттерны dto
Добавлены новые контроллеры, новые сущности, новый функционал.
TODO тесты из тз 12 спринта!

EvgenyAlexandrovichM committed:
Добавлена диаграмма базы данных

EvgenyAlexandrovichM committed:
Merge branch 'main' of github.com:EvgenyAlexandrovichM/java-filmorate into add-friends-likes

EvgenyAlexandrovichM committed:
тест

EvgenyAlexandrovichM committed:
12 спринт.1 Ревью. Подключение базы данных и реализация работы с ней.
Реализованы паттерны dto
Добавлены новые контроллеры, новые сущности, новый функционал.
TODO тесты из тз 12 спринта!

EvgenyAlexandrovichM committed:
12 спринт.1 Ревью. Подключение базы данных и реализация работы с ней.
Реализованы паттерны dto
Добавлены новые контроллеры, новые сущности, новый функционал.
TODO тесты из тз 12 спринта!

EvgenyAlexandrovichM committed:
12 спринт.1 Ревью. Подключение базы данных и реализация работы с ней.
Реализованы паттерны dto
Добавлены новые контроллеры, новые сущности, новый функционал.
TODO тесты из тз 12 спринта!

EvgenyAlexandrovichM committed:
12 спринт.1 Ревью. Подключение базы данных и реализация работы с ней.
Реализованы паттерны dto
Добавлены новые контроллеры, новые сущности, новый функционал.
TODO тесты из тз 12 спринта!

EvgenyAlexandrovichM committed:
Пользователи полностью проходят тесты в постмане.

EvgenyAlexandrovichM committed:
Реализована вся функциональность из тз12
Подключена база данных, реализованы запросы

EvgenyAlexandrovichM committed:
Реализована вся функциональность из тз12
Подключена база данных, реализованы запросы

EvgenyAlexandrovichM committed:
Реализована вся функциональность из тз12
Подключена база данных, реализованы запросы

EvgenyAlexandrovichM committed:
Реализована вся функциональность из тз12
Подключена база данных, реализованы запросы

EvgenyAlexandrovichM committed:
Все прошлые тесты на время закомментарены, поглядим!

EvgenyAlexandrovichM committed:
todo тесты junit, интеграционные
Логика прошла все постман тесты,
12 спринт. 1 ревью.

EvgenyAlexandrovichM committed:
todo тесты junit, интеграционные
Логика прошла все постман тесты,
12 спринт. 1 ревью.

EvgenyAlexandrovichM committed:
todo тесты junit, интеграционные
Логика прошла все постман тесты,
12 спринт. 1 ревью.

EvgenyAlexandrovichM committed:
todo тесты junit, интеграционные
Логика прошла все постман тесты,
12 спринт. 1 ревью.

EvgenyAlexandrovichM committed:
todo тесты junit, интеграционные
Логика прошла все постман тесты,
12 спринт. 1 ревью.

EvgenyAlexandrovichM committed:
todo тесты junit, интеграционные
Логика прошла все постман тесты,
12 спринт. 1 ревью.

EvgenyAlexandrovichM committed:
todo тесты junit, интеграционные
Логика прошла все постман тесты,
12 спринт. 1 ревью.

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/service/impl/FilmServiceImpl.java:
```diff
@@ -1,48 +1,47 @@
+        log.info("Пользователь с id {} удалил лайк с id фильму {}", userId, filmId);
     }
 
     @Override
```
ответ нужно обогатить данными по жанрам и лайкам. Только не нужно  в цикле обращаться к БД за данными, каждое соединение с БД - это затратная процедура. Заранее вытяни из БД все данные жанрам и фильмам, в потоке из них собери мапу типа Map<film.getId(), Set <Genre> > а далее уже бери жанры из этой мапы. Аналогично с лайками. И эту логику вынеси в отдельный внутренний метод класса

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/service/impl/FilmServiceImpl.java:
```diff
@@ -52,20 +51,22 @@ public List<Film> getPopularFilms(int count) {
+    public List<Film> getAll() {
         log.info("Получение всех фильмов");
-        return filmStorage.findAll();
+        return filmStorage.findAllFilms();
```
потому что здесь нужно такое же обогащение

EvgenyAlexandrovichM committed:
Изменения по ревью. Ответ популярных и всех фильмов обогащен лайками и жанрами, добавлены соответствующие методы
Также рефакторинг. Заинжектил LikeStorage.java в FilmServiceImpl.java, удалены лишние методы по логике лайков из FilmStorage.java
Удален FriendStorage.java и FriendshipDbStorage.java, вся логика по друзьям убрана в UserStorage.java

irek-valiev commented on src/main/java/ru/yandex/practicum/filmorate/storage/dal/LikeDbStorage.java:
```diff
@@ -46,4 +52,17 @@ public int countLikesByFilmId(Long filmId) {
+                        )
+                ));
+    }
+    //Есть вайб, что я зря вообще делал логику с лайками отдельно, ведь это не какая-то сущность, а просто поле фильмов
```
нет, удалять не нужно. Еще впереди групповой проект, где это приложение получит большое количество фич и выделенные лайки немного упростят реализацию некоторых из них
